packages/lesswrong/components/common/SearchBar.jsx

```diff
@@ -130,7 +130,7 @@ class SearchBar extends Component {
   }
 
   handleSearchTap = () => {
-    this.setState({inputOpen: true, searchOpen: this.state.currentQuery});
+    this.setState(prevState => ({inputOpen: true, searchOpen: !!prevState.currentQuery}));
```

Comment:
What's the reasoning for this one? Setting state directly seems simpler than defining a state-mapping.

Reply:
It's because it can cause problems when the state has changed while setState was waiting. This way you can be sure that this function always inverts the current state.